independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/spi/ResteasyReactiveResourceInfo.java

```diff
@@ -102,8 +103,10 @@ public Method getResourceMethod() {
     }
 
     public Annotation[] getParameterAnnotations(int index) {
-        // Should we cache this?
-        return getMethod().getParameterAnnotations()[index];
+        if (parameterAnnotations == null) {
```

Comment:
Given how `getMethod()` looks in the same class, there's a good chance this might need a synchronization pattern.

Reply:
Yeah, I'd say let's just skip the change to this method.

Reply:
@Postremus could you revert this specific change and squash? After that I think it's good to go.

Reply:
@gsmet I will do a squash & rebase this evening (CEST).

Reply:
done